protostar/commands/test_cairo1/test_cairo1_command.py

```diff
@@ -0,0 +1,139 @@
+""".strip(),
+                type="str",
+                value_parser="list",
+                is_positional=True,
+                default=["."],
+            ),
+            ProtostarArgument(
+                name="ignore",
+                short_name="i",
+                description=(
+                    "A glob or globs to a directory or a test suite, which should be ignored."
+                ),
+                value_parser="list",
+                type="str",
+            ),
+            ProtostarArgument(
+                name="linked-libraries",
+                value_parser="list",
+                description="Libraries to include in compilation",
+                type="path",
+            ),
+            ProtostarArgument(
+                name="no-progress-bar",
+                type="bool",
+                description="Disable progress bar.",
+            ),
+            ProtostarArgument(
+                name="exit-first",
+                short_name="x",
+                type="bool",
+                description="Exit immediately on first broken or failed test.",
+            ),
+            ProtostarArgument(
+                name="report-slowest-tests",
+                type="int",
+                description="Print slowest tests at the end.",
+                default=0,
+            ),
+            ProtostarArgument(
+                name="last-failed",
+                short_name="lf",
+                type="bool",
+                description="Only re-run failed and broken test cases.",
+            ),
+        ]
+
+    async def run(self, args: Namespace) -> TestingSummary:
+        if not vars(args).get("json"):
+            args.json = None
+        messenger = self._messenger_factory.from_args(args)
+        cache = TestCommandCache(CacheIO(self._project_root_path))
+        test_command = TestCommand(
+            project_root_path=self._project_root_path,
+            protostar_directory=self._protostar_directory,
+            project_cairo_path_builder=self._project_cairo_path_builder,
+            log_color_provider=self._log_color_provider,
+            cwd=self._cwd,
+            active_profile_name=self._active_profile_name,
+            messenger_factory=self._messenger_factory,
+        )
+        summary = await test_command.test(
+            targets=cache.obtain_targets(args.target, args.last_failed),
+            ignored_targets=args.ignore,
+            cairo_path=args.linked_libraries,
+            disable_hint_validation=False,
+            profiling=False,
+            no_progress_bar=args.no_progress_bar,
+            safe_collecting=False,
+            exit_first=args.exit_first,
+            seed=None,
+            max_steps=None,
+            slowest_tests_to_report_count=args.report_slowest_tests,
+            gas_estimation_enabled=False,
+            messenger=messenger,
+            use_cairo1_test_runner=True,
+        )
```

Comment:
Shouldn't we use this opportunity to prune some unused and irrelevant code from this command? Some redundancy would be introduced but i just wonder if you considered it. 

It's not a deal breaker for this PR though.